File: contests/abs/arc065_a.ml
open Core

let read_line () = In_channel.(input_line_exn stdin)
let cond b = if b then "YES" else "NO"

let () =
  Str.(string_match (regexp {|^\(dream\|dreamer\|erase\|eraser\)+$|})) (read_line ()) 0
  |> cond |> print_endline

File: contests/abs/arc065_a.hs
import Data.Bool (bool)
import Data.ByteString.Char8 qualified as BS
import Flow
import Text.Regex.TDFA ((=~))

main :: IO ()
main = BS.getLine >>= flip (=~) "^(dream|dreamer|erase|eraser)+$" .> bool "NO" "YES" .> putStrLn
